src/screens/Departures/state/stop-place-details-state.ts

```diff
@@ -59,6 +59,14 @@ const reducer: ReducerWithSideEffects<
         },
         async (state, dispatch) => {
           if (!action.locations) return;
+          if (action.locations.length === 0) {
+            dispatch({
+              type: 'UPDATE_STOP_DETAILS',
+              ids: action.locations,
+              result: {stopPlaces: []},
```

Comment:
This could be solved instead using the same code as below, or?

```ts
onst result = await getStopsDetails({ids: action.locations});
            dispatch({
              type: 'UPDATE_STOP_DETAILS',
              ids: action.locations,
              result: !action.locations.length ? {stopPlaces: []} : result
            });
```

Reply:
I'd also say that maybe passing the entire result here is kind of tightly coupling the dispatcher/reducer with service layer, but it's not strictly necessary to fix.

Reply:
BFF gives a 400 and goes to the catch block. SO I thought why to even call it.

Reply:
Hmm. Ok. I would say that there's kind of an odd flow here then. We're calling something we know will fail and still update the state with something that should strictly speaking be the default? I think a cleaner flow would be:

1. Sane defaults with empty lists
2. If locations, dispatch LOAD.

But I'm perfectly happy with just patching this like you have now.

Reply:
I think the empty list is actually a default already. 
But the problem occurs when 
1. We call BFF with a valid value i.e. non-empty list, 
2. Get valid data back and update stat the result
3. Call BFF again, but with an empty array which is an invalid input
4. Get error from BFF and land in catch
5. The state which now should be an empty list remains untouched leading to an incorrect state.

So an alternative would be to have this in the catch but then we have to figure out the type of error and then only do it.  😅